src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,316 @@
+                list_file_xml=glob.glob(f'{self.path_annotation}/{phrase_xml_in_dir}*.xml')
+            else:
+                list_file_xml=glob.glob(f'{self.path_annotation}/annotation/{self.kind}-{phrase_xml_in_dir}*.xml')
+            
+            if len(list_file_xml)==1:
+                self.xml_et=ET.parse(list_file_xml[0])
+                self._xml_loaded=list_file_xml[0]
+                
+            else:
+                raise ValueError(f'Search for the annotation file does not look right. len={len(list_file_xml)}')
+
+
+    def _parse_VectorList(self,name_VectorList:str,name_vector:str,str_type:str):
+        #NOTE: str type: ['datetime','scalar_integer','scalar_float','vector_integer','vector_float','str']
+        if self.xml_et is None:
+            self._load_xml_as_et()
+        
+        element_to_parse=self.xml_et.find(name_VectorList)
+        num_element=len(element_to_parse)
+
+        list_out=[None]*num_element
+        
+        if str_type=='datetime':
+            for i,elem in enumerate(element_to_parse):
+                str_elem=elem.find(name_vector).text
+                list_out[i]=datetime.datetime.strptime(str_elem,'%Y-%m-%dT%H:%M:%S.%f')
+        
+        elif str_type=='scalar_int':
+            for i,elem in enumerate(element_to_parse):
+                str_elem=elem.find(name_vector).text
+                list_out[i]=int(str_elem)
+
+        elif str_type=='scalar_float':
+            for i,elem in enumerate(element_to_parse):
+                str_elem=elem.find(name_vector).text
+                list_out[i]=float(str_elem)
+
+        elif str_type=='vector_int':
+            for i,elem in enumerate(element_to_parse):
+                str_elem=elem.find(name_vector).text
+                list_out[i]=np.array([int(strin) for strin in str_elem.split()])
+
+        elif str_type=='vector_float':
+            for i,elem in enumerate(element_to_parse):
+                str_elem=elem.find(name_vector).text
+                list_out[i]=np.array([float(strin) for strin in str_elem.split()])
+        
+        elif str_type=='str':
+            list_out=element_to_parse[0].find(name_vector).text
+        
+        else:
+            raise ValueError(f'Cannot recognize the type of the element: {str_type}')
+
+        return list_out
+
+
+    @property
+    def IPF_version(self):
+        if self._IPF_version is None:
+            if os.path.isfile(self.path_annotation):
+                str_safe=os.path.basename(self.path_annotation).replace('.zip','')
+                if self.path_annotation.endswith('.zip'):
+                    zip_in=zipfile.ZipFile(self.path_annotation,'r')
+                    str_manifest=zip_in.read(f'{str_safe}.SAFE/manifest.safe').decode()
+                    
+            elif os.path.isdir(self.path_annotation):
+                with open(f'{self.path_annotation}/manifest.safe','r') as manifest_in:
+                    str_manifest=manifest_in.read()
```

Comment:
I'm not sure if `isfile` and `isdir` can be both `False`, but I think you need an `else` condition to make sure`str_manifest` will be declared so that it can be used in the line below. Another option is to start `str_manifest` with None and check for that before using it. BTW, what happens if `self.path_annotation` is a symbolic link?

Reply:
Thanks for the comment. I've tested the `os.path.isfile()` and `os.path.isdir()` with symlink to the directory and .zip file respectively. Looks like the soft link gives `True` on both mac and linux (i.e. `aurora`).
Regardless of their behavior towards the symlinks, I agree your comment that we need to think about some situation that both of them return `False`, so I've added line to raise `ValueError`

https://github.com/seongsujeong/s1-reader/blob/7e85dd7cda8d2ff3d36779894b9e0c52365d6149/src/s1reader/s1_annotation.py#L163-L164
```python
else:
    raise ValueError(f'Cannot find the manifest.safe file using path_annotation: {self.path_annotation}')
```